WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -1149,6 +1145,10 @@ class AbstractPostListViewController: UIViewController,
     func noConnectionMessage() -> String {
         return ReachabilityUtils.noConnectionMessage()
     }
+
+    func shouldPresentAlert() -> Bool {
```

Comment:
Why do we need this? Now there we're calling handleConnectionError it should call the shouldPresentAlert in NetworkAwareUI or is there a reason we need a specific implementation in AbstractPostListViewController?

Reply:
I opted not to change this and just simply moved it to the same location as the other `NetworkAwareUI` methods. I thought that, since it overrides `NetworkAwareUI`'s `shouldPresentAlert()`, it's critical to leave it as is or I might cause a regression. 

Looking more closely, the difference is the additional `isViewOnScreen()` which is probably still important to be there for now. The `isViewOnScreen()` call will prevent us from showing a Notice in scenarios like:

1. On Site → Pages, the user refreshes the page
2. The user navigates away from the page
3. The refresh operation comes back and shows a Notice in the wrong context (!!) 

This is probably unlikely to happen though. Still, I believe we should keep this as is and look into improving `NetworkAwareUI` usage across the app later. WDYT @yaelirub?

Reply:
I'm good with that. Thanks for the elaborate response.